### PR TITLE
Expose FindRandomPointAroundCircle to Python/C + Add Unload/IsLoaded to C/Py + add pyproject.toml setup

### LINKED ISSUE #25: Use std::filesystem::path instead of std::string for paths
This is now more standard.

### LINKED ISSUE #41: Tile unload
Expose tile unloading in `pathfind` and python API.

### LINKED ISSUE #45: Expose FindRandomPointAroundCircle
https://github.com/recastnavigation/recastnavigation/blob/main/Detour/Source/DetourNavMeshQuery.cpp#L317

https://github.com/layzerar/recastlib/blob/master/python/detour/query.cpp#L460

This would help us a lot with wandering/fear movement generator.

### PR DESCRIPTION
**Pull request by gtker**
Fixes #45.
Fixes #41.
Fixes #25.

Did a few of the smaller issues while I was at it anyway.

The wheels are built every time to make sure nothing breaks, but the publishing to PyPI only happens when the commit is tagged with something starting with `v`, like `v0.1.0`.
Wheels are ready built binary packages for Python.
For the publish step to work you'll need to set the `pypi_password` secret for the repo.

I'm not sure what the official Python stance is regarding module names, but IMO it would make sense if the two current modules, `pathfind` and `mapbuild` were combined into just `namigator`. This will probably even reduce the final binary size since we're not duplicating code in the shared libraries.

I think you can also now just run `pip install .` in the repo to build and install.

### CODE REVIEW

File: pathfind/Map.cpp [hunk 1]
```diff
@@ -816,7 +863,8 @@ bool Map::ZoneAndArea(const math::Vertex& position, unsigned int& zone,
         {position.X, position.Y, tile->second->m_bounds.getMinimum().Z}};
 
     unsigned int localZone, localArea;
-    auto const rayResult = RayCast(ray, tiles, &localZone, &localArea);
+    const auto doodads = false;
```
Comment (namreeb) [inline pathfind/Map.cpp:866]: This doesn't seem to be referenced anywhere else.  You can just put `false` into the `RayCast()` call.

File: pathfind/pathfind_c_bindings.cpp [hunk 2]
```diff
@@ -91,6 +91,32 @@ PathfindResultType pathfind_load_adt_at(pathfind::Map* const map, float x, float
     }
 }
 
+PathfindResultType pathfind_unload_adt(pathfind::Map* const map, int x, int y) {
+    try {
+        map->UnloadADT(x, y);
+    }
+    catch (...) {
+        return static_cast<PathfindResultType>(Result::UNKNOWN_EXCEPTION);
```
Comment (namreeb) [inline pathfind/pathfind_c_bindings.cpp:99]: It looks like we're discarding valuable information about the failure here.

File: pathfind/pathfind_c_bindings.cpp [hunk 3]
```diff
@@ -91,6 +91,32 @@ PathfindResultType pathfind_load_adt_at(pathfind::Map* const map, float x, float
...
+PathfindResultType pathfind_is_adt_loaded(pathfind::Map* const map, int x, int y, uint8_t* const loaded) {
+    try {
+        if (map->IsADTLoaded(x, y)) {
+            *loaded = 1;
+        } else{
+            *loaded = 0;
+        }
+    }
+    catch (...) {
+        return static_cast<PathfindResultType>(Result::UNKNOWN_EXCEPTION);
```
Comment (namreeb) [inline pathfind/pathfind_c_bindings.cpp:114]: It looks like we're discarding valuable information about the failure here, too.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+[build-system]
+requires = ["scikit-build-core[pyproject]>=0.3.3"]
```
Comment (namreeb) [inline pyproject.toml:2]: Why is scikit required?

File: utility/MathHelper.cpp [hunk 5]
```diff
@@ -148,6 +148,19 @@ void Convert::VertexToRecast(const Vector3& input, float* output)
     output[2] = -x;
 }
 
+void Convert::RecastToVertex(const float* const input, Vector3& output)
```
Comment (namreeb) [inline utility/MathHelper.cpp:151]: How is this different than `VertexToWow`?

**Review by namreeb — CHANGES_REQUESTED** [review]
Please replace 'NAMIGATOR_ONLY_BUILD_MINIMAL_PYTHON' with options to build python bindings and c bindings. The 'minimal python build' as I understand it would have those on and off, respectively.

**Comment by gtker** [comment]
I'll take a look a little later.

> Please replace 'NAMIGATOR_ONLY_BUILD_MINIMAL_PYTHON' with options to build python bindings and c bindings. The 'minimal python build' as I understand it would have those on and off, respectively.

`NAMIGATOR_ONLY_BUILD_MINIMAL_PYTHON` also disables building `MapViewer` for Windows, do you want a separate setting that defaults to true to toggle building that?

File: pyproject.toml (hunk 4 above)
Comment (gtker) [reply pyproject.toml:2]: Scikit handles the cmake to python integration. Even though it's weirdly named [scikit-build-core](https://github.com/scikit-build/scikit-build-core) is a build tool and not a scikit library.

File: utility/MathHelper.cpp (hunk 5 above)
Comment (gtker) [reply utility/MathHelper.cpp:151]: It isn't, I missed that `VertexToWow` existed.

File: pathfind/pathfind_c_bindings.cpp (hunk 3 above)
Comment (gtker) [reply pathfind/pathfind_c_bindings.cpp:114]: That's true.

**Comment by namreeb** [comment]
> NAMIGATOR_ONLY_BUILD_MINIMAL_PYTHON also disables building MapViewer for Windows, do you want a separate setting that defaults to true to toggle building that?

How about the following options:
`BUILD_VIEWER` -- default to on for Windows, else off
`BUILD_PYTHON` -- default to on
`BUILD_C` -- default to on

**Review by namreeb — APPROVED** [review]

**Pull request merged**